pkg/services/publicdashboards/database/database.go

```diff
@@ -283,15 +283,15 @@ func (d *PublicDashboardStoreImpl) Delete(ctx context.Context, uid string) (int6
 	return affectedRows, err
 }
 
-func (d *PublicDashboardStoreImpl) FindByDashboardFolder(ctx context.Context, dashboard *dashboards.Dashboard) ([]*PublicDashboard, error) {
-	if dashboard == nil || !dashboard.IsFolder {
+func (d *PublicDashboardStoreImpl) FindByFolder(ctx context.Context, orgId int64, folderUid string) ([]*PublicDashboard, error) {
+	if orgId == 0 || folderUid == "" {
 		return nil, nil
 	}
 
 	var pubdashes []*PublicDashboard
 
 	err := d.sqlStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		return sess.SQL("SELECT * from dashboard_public WHERE (dashboard_uid, org_id) IN (SELECT uid, org_id FROM dashboard WHERE folder_id = ?)", dashboard.ID).Find(&pubdashes)
+		return sess.SQL("SELECT * from dashboard_public WHERE (dashboard_uid, org_id) IN (SELECT uid, org_id FROM dashboard WHERE org_id = ? AND folder_uid = ?)", orgId, folderUid).Find(&pubdashes)
 	})
```

Comment:
in the dashboard table, if the dashboard is a folder, the `dashboard_uid` is equivalent to the `folder_uid` in the folders table, the `folder_uid` in the dashboard table is the uid of the parent folder, it a bit confusing, but that's why, in the query the value used should be dashboard_uid instead of folder_uid

Reply:
I can see in the funtion call that the function is using a `dashboard_uid` of a dashboardfolder for this call which is correct, I am just trying to think how to make the naming less confusing

Reply:
I know it's confusing 😅 , so I decided to abstract everything inside this method using the correct parameter name. The purpose is to find resources by a `folderUid` independent of how the caller populates or obtains that value.

Reply:
what about `dashboardfolder_uid`, or `dashboard_uid`, I just want to make it clear that it is not the `dashboard.folderUID`